### PR TITLE
Read and write dataframes from and to NDJSON files

### PR DESCRIPTION
**Pull request by philss**
[NDJSON](https://github.com/ndjson/ndjson-spec) is an useful format to store multiple JSON documents in a single
file, separating them by new lines.

Closes https://github.com/elixir-nx/explorer/issues/139

### CODE REVIEW

File: lib/explorer/data_frame.ex [hunk 1]
```diff
@@ -153,6 +153,44 @@ defmodule Explorer.DataFrame do
     end
   end
 
+  @doc """
+  Read a file of JSON objects or lists separated by new lines
+
+  ## Options
+
+    * `with_batch_size` - Sets the batch size for reading rows.
+    This value may have significant impect in performance, so adjust it for your needs (default: `1000`).
```
Comment (cigrainger) [inline lib/explorer/data_frame.ex:162]:
```suggestion
    This value may have significant impact in performance, so adjust it for your needs (default: `1000`).
```

**Review by cigrainger — COMMENTED** [review]
Just caught the one typo. Otherwise: wonderful! Thanks @philss :)

**Comment by philss** [comment]
@cigrainger thanks! :purple_heart: 
I will fix the conflicts and the typo manually.

**Review by cigrainger — APPROVED** [review]
Wonderful!

**Pull request merged**